### PR TITLE
[fix-eslint-error] Fix error in utils, models, configs folder (except…

### PR DESCRIPTION
**Pull request by trivm2472**
… for refreshToken.model)

### CODE REVIEW

File: back-end/src/controllers/authentication.controller.ts [hunk 1]
```diff
@@ -12,14 +12,22 @@ import { User } from '../services/users.service';
...
+        success: false,
+        error: 'Username already exist',
+        errorCode: 'SIGNUP009',
+      };
+    } else if (user.email == data.email) {
+      return {
+        success: false,
+        error: 'Email already exist',
+        errorCode: 'SIGNUP010',
+      };
```
Comment (vuhah) [inline back-end/src/controllers/authentication.controller.ts:30]: t nhớ hôm qua thống nhất đặt errorCode giống Sil là dùng '_' á. kiểu như SIGN_UP_009 á

File: back-end/src/controllers/authentication.controller.ts [hunk 2]
```diff
@@ -12,14 +12,22 @@ import { User } from '../services/users.service';
...
-      return { success: false, error: 'Username already exist' };
+    const user = await UsersService.find(data);
+    if (!user) {
+      return { success: true };
     }
-
-    const checkEmail = await UsersService.find({ email: data.email });
-    if (checkEmail) {
-      return { success: false, error: 'Email already exist' };
+    if (user.username == data.username) {
```
Comment (caohaisil201) [inline back-end/src/controllers/authentication.controller.ts:19]: Can use '===' instead of '=='

File: back-end/src/models/refreshToken.model.ts [hunk 3]
```diff
@@ -1,9 +1,8 @@
 /* eslint-disable */
```
Comment (caohaisil201) [inline back-end/src/models/refreshToken.model.ts:1]: Can change it to /* eslint-disable camelcase */

Comment (caohaisil201) [reply back-end/src/models/refreshToken.model.ts:1]: I wrong, we should delete it

File: back-end/src/models/users.model.ts [hunk 4]
```diff
@@ -89,18 +88,19 @@ const registerSchema = Joi.object({
...
+    .pattern(new RegExp('^[a-zA-Z0-9_-]+$'))
+    .required(),
+  password: Joi.string()
+    .min(8)
+    .max(16)
+    .pattern(new RegExp('^[a-zA-Z0-9_-]+$'))
+    .required(),
+});
 
+module.exports = {
```
Comment (caohaisil201) [inline back-end/src/models/users.model.ts:102]: We should change it to es6

File: back-end/src/utils/Mail.utils.ts [hunk 5]
```diff
@@ -1,46 +1,43 @@
...
-
+import nodemailer from 'nodemailer';
+import SMTPTransport from 'nodemailer/lib/smtp-transport';
+import { FORGOT_PASSWORD } from './Helper.utils';
+/* eslint-disable no-debugger, no-console */
 
 export class Email {
   private transporter = nodemailer.createTransport({
     service: 'gmail',
     auth: {
```
Comment (caohaisil201) [inline back-end/src/utils/Mail.utils.ts:9]: No inline comment, user and pass should be moved to .env

File: back-end/src/utils/Mail.utils.ts [hunk 6]
```diff
@@ -1,46 +1,43 @@
...
       user: 'bozuman2022@gmail.com',
-      pass: 'qafiidxzwzimdcku' //Real password is bozuman123
-    }
+      pass: 'qafiidxzwzimdcku', //Real password is bozuman123
+    },
   });
 
   //SIL: change this function for active account and sendcode
```
Comment (caohaisil201) [inline back-end/src/utils/Mail.utils.ts:15]: delete my comment

File: back-end/src/utils/Mail.utils.ts [hunk 7]
```diff
@@ -1,46 +1,43 @@
...
-  public sendEmail = (emailAddress: any, token: any, type: string) => {
-    
+  public sendEmail = (emailAddress: string, token: string, type: string) => {
     const mailOptions = {
       from: 'bozuman2022@gmail.com',
       to: emailAddress,
-      subject: 'Activatetion code',
-      html: `<p>Click <a href="${process.env.PROJECT_DOMAIN}/api/auth/activate_account/${token}>here</a> to activate your account</p>`
+      subject: 'Activation code',
+      html: `<p>Click <a href="${process.env.PROJECT_DOMAIN || ''}/api/auth/activate_account/${token}>here</a> to activate your account</p>`,
```
Comment (caohaisil201) [inline back-end/src/utils/Mail.utils.ts:26]: || {''} là sao nhỉ

**Pull request closed without merging**